deep_qa/data/instances/sequence_tagging/pretokenized_tagging_instance.py

```diff
@@ -0,0 +1,61 @@
+from typing import List
+
+import numpy
+from overrides import overrides
+
+from .tagging_instance import TaggingInstance
+from ...data_indexer import DataIndexer
+
+class PreTokenizedTaggingInstance(TaggingInstance):
+    """
+    This is a ``TaggingInstance`` where the text has been pre-tokenized.  Thus the ``text`` member
+    variable here is actually a ``List[str]``, instead of a ``str``.
+
+    When using this ``Instance``, you `must` use the ``NoOpWordSplitter`` as well, or things will
+    break.  You probably also do not want any kind of filtering (though stemming is ok), because
+    only the words will get filtered, not the labels.
+    """
+    def __init__(self, text: List[str], label: List[str], index: int=None):
+        super(PreTokenizedTaggingInstance, self).__init__(text, label, index)
+
+    @classmethod
+    @overrides
+    def read_from_line(cls, line: str, default_label: bool=None):
+        """
+        Reads a ``PreTokenizedTaggingInstance`` from a line.  The format has one of two options:
+
+        1. [example index][token1]###[tag1][tab][token2]###[tag2][tab]...
+        2. [token1]###[tag1][tab][token2]###[tag2][tab]...
+
+        default_label is ignored, but we keep the argument to match the interface.
+        """
+        fields = line.split("\t")
+
+        if fields[0].isdigit():
+            index = int(fields[0])
+            fields = fields[1:]
+        else:
+            index = None
+        tokens = []
+        tags = []
+        for field in fields:
+            token, tag = field.split("###")
+            tokens.append(token)
+            tags.append(tag)
+        return cls(tokens, tags, index)
+
+    @overrides
+    def words_in_label(self):
+        return [tag for tag in self.label]
```

Comment:
In this specific case, these are tokens, right? It confused me a bit that they were called tags rather than tokens. I would expect the tag to be in `{B, I, O}`.